bin/tests/cli.rs

```diff
@@ -391,3 +396,36 @@ fn test_directory_symlinks_delete() {
     common::assert_agent_running(&mut agent_handle);
     agent_handle.kill().expect("Could not kill process");
 }
+
+#[test]
+#[cfg_attr(not(feature = "integration_tests"), ignore)]
+#[cfg_attr(not(target_os = "linux"), ignore)]
+fn test_journald_support() {
+    assert_eq!(journal::print(6, "Sample info"), 0);
+    sleep(Duration::from_millis(1000));
+    let dir = "/var/log/journal";
+    let mut agent_handle = common::spawn_agent(AgentSettings {
+        log_dirs: "/var/log/",
+        journald_dirs: Some(dir),
+        ..Default::default()
+    });
+
+    let mut stderr_reader = BufReader::new(agent_handle.stderr.as_mut().unwrap());
+
+    common::wait_for_file_event(
+        "monitoring journald path",
+        &PathBuf::from(dir),
+        &mut stderr_reader,
+    );
+
+    for i in 0..10 {
+        journal::print(1, format!("Sample alert {}", i).as_str());
+        journal::print(6, format!("Sample info {}", i).as_str());
+    }
+
+    common::wait_for_event("sending journald line", &mut stderr_reader);
+    common::wait_for_event("sending http request", &mut stderr_reader);
```

Comment:
Not very elegant...

Once we have the mock ingester as a dev dependency we can verify what was received.